components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/TLConnectionStore.java

```diff
@@ -75,6 +78,9 @@ public static void commitNonXAConns() {
             if (!conn.isXA()) {
                 try {
                     conn.commit();
+                    if (log.isDebugEnabled()) {
+                        log.debug("Committing non-XA connection. ThreadID:" + Thread.currentThread().getId());
```

Comment:
Shall we print the "conn" object as well, which will invoke it's "toString", so do please override toString in "DataServiceConnection" classes, to print something suitable, for example, in RDBMS ones, it's Connection objects.